sdk/scheduler/src/main/java/com/mesosphere/sdk/api/PodResource.java

```diff
@@ -277,22 +391,48 @@ public int compare(TaskInfoAndStatus a, TaskInfoAndStatus b) {
         }
     }
 
-    private static JSONArray getStatusesJson(List<TaskInfoAndStatus> tasks) {
+    private static JSONArray getStatusesJson(StateStore stateStore, List<TaskInfoAndStatus> tasks) {
         JSONArray jsonPod = new JSONArray();
         for (TaskInfoAndStatus task : tasks) {
             JSONObject jsonTask = new JSONObject();
             jsonTask.put("id", task.getInfo().getTaskId().getValue());
             jsonTask.put("name", task.getInfo().getName());
             if (task.hasStatus()) {
-                jsonTask.put("state", task.getStatus().get().getState().toString());
-            } else {
-                jsonTask.put("state", "No state defined");
+                jsonTask.put("state",
+                        getSDKTaskState(stateStore, task.getInfo().getName(), task.getStatus().get().getState()));
+            }
+            try {
+                jsonTask.put("type", new TaskLabelReader(task.getInfo()).getType());
+            } catch (TaskException e) {
+                // no type found, just omit type field from response
             }
             jsonPod.put(jsonTask);
         }
         return jsonPod;
     }
 
+    private static String getSDKTaskState(StateStore stateStore, String taskName, Protos.TaskState mesosState) {
```

Comment:
Javadoc please.  Comment on line 417 should probably move up.

Reply:
Removed this, it was temporary and is being replaced again in the following PR (currently the `stop-pods-podstatus` branch).